site/docs/custom-catalog.md

```diff
@@ -127,8 +137,111 @@ public class CustomCatalog extends BaseMetastoreCatalog {
     // Example service to rename table
     CustomService.renameTable(from.namepsace().level(0), from.name(), to.name());
   }
+
+  // implement this method to read catalog name and properties during initialization
+  public void initialize(String name, Map<String, String> properties) {
+  }
 }
 ```
+
+Catalog implementations can be dynamically loaded in most compute engines.
+For Spark and Flink, you can specify the `catalog-impl` catalog property to load it.
+Read the [Configuration](../configuration/#catalog-properties) section for more details.
+For MapReduce, implement `org.apache.iceberg.mr.CatalogLoader` and set Hadoop property `iceberg.mr.catalog.loader.class` to load it.
+If your catalog must read Hadoop configuration to access certain environment properties, make your catalog implement `org.apache.hadoop.conf.Configurable`.
+
+### Custom file IO implementation
+
+Extend `FileIO` and provide implementation to read and write data files
+
+Example:
+```java
+public class CustomFileIO implements FileIO {
```

Comment:
This isn't really needed for a custom catalog, and it is an option for the built-in catalogs. I think maybe we should move this to a doc on customizing tables, instead of here. Or, maybe just make it more clear that this isn't required. I think it is somewhat common for people to want to customize catalogs, and I don't want to make it seem like it is harder than it already is.

Reply:
Actually, I think this is fine. We may want to clarify some of the wording, but I think this page is turning into general docs for customizing tables.

Reply:
Yes, maybe a more generic title like `Customization` would make more sense now.